pinot-core/src/main/java/org/apache/pinot/core/transport/ChannelHandlerFactory.java

```diff
@@ -61,14 +64,22 @@ public static ChannelHandler getLengthFieldPrepender() {
    * The {@code getClientTlsHandler} return a Client side Tls handler that encrypt and decrypt everything.
    */
   public static ChannelHandler getClientTlsHandler(TlsConfig tlsConfig, SocketChannel ch) {
-    return TlsUtils.buildClientContext(tlsConfig).newHandler(ch.alloc());
+    // Make a copy of the TlsConfig because the TlsConfig is mutable, when the TlsConfig is used as the key of the
+    // CLIENT_SSL_CONTEXTS_CACHE, the TlsConfig should not be changed.
+    TlsConfig tlsConfigCopy = new TlsConfig(tlsConfig);
```

Comment:
what's the frequency of this call and object copy?

Reply:
The frequency should be low, only at the time when the channel is created. It is the same as how many times the channel should be created.

Anyway, changed the map to use hashCode as the key, so the object copy can be avoided.